3 Domain layer/CandidatesEvaluator.Contract/Handlers/IQueryHandler.cs

```diff
@@ -0,0 +1,14 @@
+﻿using CandidateEvaluator.Contract.Queries;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CandidateEvaluator.Contract.Handlers
+{
+    public interface IQueryHandler<TQuery, TResult> where TQuery : IQuery<TResult>
```

Comment:
Make it both [variant and contravariant](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/covariance-contravariance/creating-variant-generic-interfaces) with in/out operators like below:
`public interface IQueryHandler<in TQuery, out TResult> where TQuery : IQuery<TResult>`

Reply:
![image](https://user-images.githubusercontent.com/10583682/49898030-16ece080-fe58-11e8-8d0a-111f6bd960ba.png)
Nope :(

Reply:
`IQuery<out T>` ?